app/uk/gov/hmrc/serviceconfigs/persistence/DeploymentConfigSnapshotRepository.scala

```diff
@@ -66,16 +79,118 @@ class DeploymentConfigSnapshotRepository @Inject()(
       .toFuture
       .map(_ => ())
 
+  private [persistence] def removeLatestFlagForNonDeletedSnapshotsInEnvironment(environment: Environment): Future[Unit] =
+    collection
+      .updateMany(
+        filter = and(
+          equal("deploymentConfig.environment", environment.asString),
+          equal("latest", true),
+          equal("deleted", false),
+        ),
+        update = set("latest", false)
+      ).toFuture().map(_ => ())
+
+  private [persistence] def removeLatestFlagForServiceInEnvironment(serviceName: String, environment: Environment): Future[Unit] =
+    collection
+      .updateMany(
+        filter = and(
+          equal("deploymentConfig.name", serviceName),
+          equal("deploymentConfig.environment", environment.asString),
+        ),
+        update = set("latest", false)
+      ).toFuture().map(_ => ())
+
+
   def populate(date: Instant): Future[Unit] = {
 
     def forEnvironment(environment: Environment): Future[Unit] = {
       for {
         deploymentConfigs <- deploymentConfigRepository.findAll(environment)
-        snapshots         =  deploymentConfigs.map(DeploymentConfigSnapshot(date, _))
-        _                 <- collection.insertMany(snapshots).toFuture()
+        latestSnapshots   <- latestSnapshotsInEnvironment(environment)
+        planOfWork        =  PlanOfWork.fromLatestSnapshotsAndCurrentDeploymentConfigs(latestSnapshots.toList, deploymentConfigs.toList, date)
+        _                 <- executePlanOfWork(planOfWork, environment)
       } yield ()
     }
 
     Environment.values.foldLeftM(())((_, env) => forEnvironment(env))
   }
+
+  def executePlanOfWork(planOfWork: PlanOfWork, environment: Environment): Future[Unit] = {
+    logger.debug(s"Processing `DeploymentConfigSnapshot`s for ${environment.asString}")
+
+    def reintroduceServiceSnapshot(deploymentConfigSnapshot: DeploymentConfigSnapshot) = {
+      logger.debug(s"Creating a snapshot for reintroduced service: $deploymentConfigSnapshot")
+      for {
+        _ <- removeLatestFlagForServiceInEnvironment(
+          deploymentConfigSnapshot.deploymentConfig.name,
+          deploymentConfigSnapshot.deploymentConfig.environment
+        )
+        _ <- collection.insertOne(deploymentConfigSnapshot).toFuture()
+      } yield ()
+    }
+
+    for {
+      _ <- if (planOfWork.snapshots.nonEmpty) removeLatestFlagForNonDeletedSnapshotsInEnvironment(environment) else Future.unit
+      _ <- collection.insertMany(planOfWork.snapshots).toFuture()
+      _ <- planOfWork.snapshotServiceReintroductions.foldLeftM(())((_, ssr) => reintroduceServiceSnapshot(ssr))
```

Comment:
maybe add a comment - "reintroductions are treated separately to ensure latest flag is removed from previously deleted entries - not covered by above bulk flag removal"

Or could you do
```scala
for {
      _ <- if (planOfWork.snapshots.nonEmpty) removeLatestFlagForNonDeletedSnapshotsInEnvironment(environment) else Future.unit
      _ <- planOfWork.snapshotServiceReintroductions.foldLeftM(())((_, ssr) => removeLatestFlagForServiceInEnvironment)
      _ <- collection.insertMany(planOfWork.snapshots ++ planOfWork.snapshotServiceReintroductions).toFuture()
} yield ()
```
`removeLatestFlagForServiceInEnvironment` could even take a list of services, and do them all in one update with `in(, serviceNames)`